download_test_vectors.sh

```diff
@@ -30,28 +33,72 @@ cleanup() {
 trap cleanup SIGINT
 
 dl_version() {
-	[[ -z "$1" ]] && { echo "usage: dl_version() vX.Y.Z"; exit 1; }
+	[[ -z "$1" ]] && { echo "usage: dl_version() vX.Y.Z | nightly"; exit 1; }
 	version="$1"
 
 	mkdir -p "tarballs/${version}"
 	pushd "tarballs/${version}" >/dev/null
-	for flavour in "${FLAVOURS[@]}"; do
-		if [[ ! -e "${flavour}.tar.gz" ]]; then
-			echo "Downloading: ${version}/${flavour}.tar.gz"
-			curl --location --remote-name --silent --show-error --retry 3 --retry-all-errors \
-				"https://github.com/ethereum/consensus-spec-tests/releases/download/${version}/${flavour}.tar.gz" \
-				|| {
-					echo "Curl failed. Aborting"
-					rm -f "${flavour}.tar.gz"
+
+	if [[ "$version" == "nightly" ]]; then
+		if [[ -z "${GITHUB_TOKEN:-}" ]]; then
+			echo "Error GITHUB_TOKEN is not set"
+			exit 1
+		fi
+
+		repo="ethereum/consensus-specs"
+		api="https://api.github.com"
+		auth_header="Authorization: token ${GITHUB_TOKEN}"
+
+		run_id=$(curl -s -H "${auth_header}" \
+			"${api}/repos/${repo}/actions/workflows/generate_vectors.yml/runs?branch=dev&status=success&per_page=1" \
+			| jq -r '.workflow_runs[0].id')
```

Comment:
Would need to check on the universal availability of `jq` too. Of course it's installable, it's been made available everywhere, but whether it's part of default environments. Maybe, wouldn't be surprised.

Reply:
Good idea. Add a check for these.